// nest-bcrypt/dist/bcrypt.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var BcryptModule_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.BcryptModule = void 0;
const common_1 = require("@nestjs/common");
const bcrypt_service_1 = require("./bcrypt.service");
const symbols_1 = require("./symbols");
let BcryptModule = BcryptModule_1 = class BcryptModule {
    static register(options) {
        return {
            module: BcryptModule_1,
            providers: [
                {
                    provide: symbols_1.BCRYPT_MODULE_OPTIONS,
                    useValue: options,
                },
                bcrypt_service_1.BcryptService,
            ],
            exports: [bcrypt_service_1.BcryptService],
        };
    }
};
BcryptModule = BcryptModule_1 = __decorate([
    (0, common_1.Module)({
        providers: [bcrypt_service_1.BcryptService],
    })
], BcryptModule);
exports.BcryptModule = BcryptModule;
